src/arrayMethodJoin.js

```diff
@@ -5,8 +5,22 @@
  *
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function() {
-    // write code here
+  [].__proto__.join2 = function(arg) {
+    return this.reduce((str, item) => {
+      if (arg === undefined) {
+        arg = ',';
+      }
+      if (arg === null) {
+        arg += '';
```

Comment:
is this the same as 'arg' = ''? Maybe let's change to `arg = ''` for clarity?

Reply:
I didn`t understant your comment. The check (arg === null) needed when the function called with argument null. In this case arg must become string 'null'.